client/components/time-since/index.jsx

```diff
@@ -1,40 +1,36 @@
 import moment from 'moment';
-import { PureComponent } from 'react';
-import humanDate from 'calypso/lib/human-date';
-import { Interval, EVERY_TEN_SECONDS } from 'calypso/lib/interval';
-import smartSetState from 'calypso/lib/react-smart-set-state';
+import { useState, useEffect } from 'react';
+import getHumanDate from 'calypso/lib/human-date';
+import { useInterval, EVERY_TEN_SECONDS } from 'calypso/lib/interval';
 
-export default class TimeSince extends PureComponent {
-	smartSetState = smartSetState;
+function TimeSince( { className, date, dateFormat } ) {
+	const [ humanDate, setHumanDate ] = useState( '' );
+	const [ fullDate, setFullDate ] = useState( '' );
```

Comment:
`fullDate` doesn't need to be in state. It depends only on `date` and can be directly derived (and memoized) in the render function:
```js
const fullDate = useMemo( () => moment( date ).format( 'llll' ), [ date ] );
```
where `date` comes from `props`.

`humanDate` is different because it also internally depends on "now" time and needs to be recomputed regularly, like every 10 seconds here.